### ISSUE TITLE
How to "retrigger" snippet

### ISSUE THREAD

**Issue by castroavila** [SirVer/ultisnips#1478]
**Expected behavior:**

Be able to "retrigger" the snippet.

**Actual behavior:**
Usually I use  (for Python):
`class`  + `Ctrl+E` to trigger the snippet and it expands as expected:
```
class MyClass(object):

    """Docstring for MyClass. """

    def __init__(self, a, b):
        """TODO: to be defined.

        Parameters
        ----------
        a : TODO
        b : TODO


        """
        self.a = a
        self.b = b
```
Each time I add a parameter in `__init__,` a new instance `self.b`is created.  However if I leave the `__init__` scope and return afterwards, if I add  a new parameter `c` in it, no  new instance  `self.c` is added nor added to parameters list: The snippet is deactivated somehow.

Is there any way to activate again the snippet?

**Steps to reproduce**
```
let g:UltiSnipsExpandTrigger="<c-e>"
let g:UltiSnipsJumpForwardTrigger="<c-b>"
let g:UltiSnipsListSnippets="<c-l>"
let g:UltiSnipsJumpBackwardTrigger="<c-z>"
let g:UltiSnipsEditSplit="vertical"
let g:ultisnips_python_style="numpy"
```

-----

- **Operating System**:  Mac Big Sur
- **Vim Version**: VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Apr 30 2022 15:10:27)
macOS version - x86_64
- **UltiSnips Version**: 3.2 
- **Python inside Vim**:  3.10.4 
- **Docker repo/vimrc**:  Not available

**Comment by SirVer** [comment]
No, this is working as intended. The reason is that UltiSnips needs to do a very expensive diff operation every time a snippet is active and the cursor is moved to figure out how the buffer changed - Vim has no affordances to get this in any other way. To only do this for as short as possible, as soon as you exit the snippet scope, the snippet terminates. It also ends when you hit `$0`. Unfortunately I did not find a way to keep snippets always active without freezing Vim to a hold.

**Comment by castroavila** [comment]
Thanks for the reply and explanation. 

Any suggestion as workaround? Because usually when I develop a code,  frequently  arguments in a function/constructor are added/deleted  and I found very handy the kind of functionality the snippet provides: instantiation, add to the parameters list, etc..

**Comment by SirVer** [comment]
@castroavila You can try snippet actions, i.e. doing something on jump to update code. It will be a bit janky I think.

https://github.com/SirVer/UltiSnips/blob/master/doc/UltiSnips.txt#L1616
